modules/jdbc/src/main/java/org/apache/ignite/internal/jdbc2/JdbcConnection2.java

```diff
@@ -0,0 +1,696 @@
+        ensureNotClosed();
+
+        throw new SQLFeatureNotSupportedException("Auto generated keys are not supported.");
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public PreparedStatement prepareStatement(String sql, int resSetType,
+            int resSetConcurrency) throws SQLException {
+        return prepareStatement(sql, resSetType, resSetConcurrency, HOLD_CURSORS_OVER_COMMIT);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public PreparedStatement prepareStatement(String sql, int resSetType, int resSetConcurrency,
+            int resSetHoldability) throws SQLException {
+        ensureNotClosed();
+
+        checkCursorOptions(resSetType, resSetConcurrency, resSetHoldability);
+
+        if (sql == null) {
+            throw new SQLException("SQL string cannot be null.");
+        }
+
+        throw new UnsupportedOperationException();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public PreparedStatement prepareStatement(String sql, int[] colIndexes) throws SQLException {
+        ensureNotClosed();
+
+        throw new SQLFeatureNotSupportedException("Auto generated keys are not supported.");
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public PreparedStatement prepareStatement(String sql, String[] colNames) throws SQLException {
+        ensureNotClosed();
+
+        throw new SQLFeatureNotSupportedException("Auto generated keys are not supported.");
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String nativeSQL(String sql) throws SQLException {
+        ensureNotClosed();
+
+        Objects.requireNonNull(sql);
+
+        return sql;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void setAutoCommit(boolean autoCommit) throws SQLException {
+        ensureNotClosed();
+
+        if (autoCommit != this.autoCommit) {
+            this.autoCommit = autoCommit;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean getAutoCommit() throws SQLException {
+        ensureNotClosed();
+
+        return autoCommit;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void commit() throws SQLException {
+        ensureNotClosed();
+
+        if (autoCommit) {
+            throw new SQLException("Transaction cannot be committed explicitly in auto-commit mode.");
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void rollback() throws SQLException {
+        ensureNotClosed();
+
+        if (autoCommit) {
+            throw new SQLException("Transaction cannot be rolled back explicitly in auto-commit mode.");
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void rollback(Savepoint savepoint) throws SQLException {
+        ensureNotClosed();
+
+        if (savepoint == null) {
+            throw new SQLException("Invalid savepoint.");
+        }
+
+        if (autoCommit) {
+            throw new SQLException("Auto-commit mode.");
```

Comment:
I believe the message should be consistent with the method above

Reply:
looks like you replied to wrong comment or something. In `rollback` without params you throw exception with message `Transaction cannot be rolled back explicitly in auto-commit mode.`. I believe we should throw exception with similar messages in similar cases. BTW, deriving messages to constants may help to avoid such problems